### PR TITLE
cp-282: Fixed Load Mobile Meditation Entries

### PR DESCRIPTION
**Pull request by iamAlinaaa**
Fixed loading Meditation entries on the appropriate screen

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/111873795/86a2df26-c215-401f-a45d-9dc89bdf0fae

### CODE REVIEW

File: mobile/src/screens/meditation-home/libs/constants.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const mockedData = [
+  { id: 1, title: 'Meditation' },
+  { id: 2, title: 'Relaxing sounds' },
+  { id: 3, title: 'Breath exercise' },
+];
+export { mockedData };
```
Comment (danilittaa) [inline mobile/src/screens/meditation-home/libs/constants.ts]: I think there's no need to name this array 'mocked' anymore because these are the actual data

**Review by OlenaYatseniuk — DISMISSED** [review]

**Review by OlesiaTr — DISMISSED** [review]

File: mobile/src/screens/meditation-home/libs/constants.ts (hunk 1 above)
Comment (iamAlinaaa) [reply mobile/src/screens/meditation-home/libs/constants.ts]: @danilittaa please, show me actual data for meditation topics 😊 i know there are actual data for entries , but i havent seen topics

Comment (danilittaa) [reply mobile/src/screens/meditation-home/libs/constants.ts]: As I understand it, we won't be receiving this data from the backend. These data will simply be static. I checked the frontend, and here's what this array is called there: 
image
You can temporarily leave only "Meditation" in this array.

Comment (iamAlinaaa) [reply mobile/src/screens/meditation-home/libs/constants.ts]: @danilittaa okay, i can call it navigationItems, but QA asked to make these three topics here, thats why i did it like that. i think it needs more investigation

File: mobile/src/screens/meditation-home/libs/constants.ts
Comment (v0ldemar01) [inline mobile/src/screens/meditation-home/libs/constants.ts:1]: libs/constants/constants

File: mobile/src/screens/meditation-home/libs/constants.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const navigationItems = [
```
Comment (v0ldemar01) [inline mobile/src/screens/meditation-home/libs/constants.ts]: Follow quality criteria

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: mobile/src/screens/meditation-list/libs/constants.ts [hunk 3]
```diff
@@ -1,10 +1,3 @@
-const mockedData = [
-  { id: 0, title: '1Meditation for deep sleep', duration: 10 },
-  { id: 1, title: 'Breathing meditation', duration: 100 },
-  { id: 2, title: '2Meditation for deep sleep', duration: 99 },
-  { id: 3, title: '3Meditation for deep sleep', duration: 10 },
-  { id: 4, title: '4Breathing meditation', duration: 100 },
-  { id: 5, title: '5Meditation for deep sleep', duration: 99 },
-];
+const MOCKED_DURATION = 10;
```
Comment (v0ldemar01) [inline mobile/src/screens/meditation-list/libs/constants.ts]: Should we use it for now?

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: mobile/src/screens/meditation-list/libs/constants.ts (hunk 3 above)
Comment (iamAlinaaa) [reply mobile/src/screens/meditation-list/libs/constants.ts]: @v0ldemar01 i didnt see duration of track in meditation entry object for now

File: mobile/src/screens/meditation-home/libs/constants.ts (hunk 2 above)
Comment (iamAlinaaa) [reply mobile/src/screens/meditation-home/libs/constants.ts]: @v0ldemar01 could you please explain whats wrong with it ? we have the same name on frontend and B18. Arrays are named plural nouns.

File: mobile/src/screens/meditation-list/libs/constants.ts (hunk 3 above)
Comment (iamAlinaaa) [reply mobile/src/screens/meditation-list/libs/constants.ts]: @v0ldemar01 i have troubles with duration in react native track player it doesnt work properly, and i need to link another library, but how to link to our project ?

Comment (danilittaa) [reply mobile/src/screens/meditation-list/libs/constants.ts]: @iamAlinaaa , What's wrong with duration in this library? Ask me, I'll help you

File: mobile/src/screens/meditation-home/libs/constants.ts (hunk 2 above)
Comment (danilittaa) [reply mobile/src/screens/meditation-home/libs/constants.ts]: @iamAlinaaa B22. Constant names are written in capital letters.

Comment (iamAlinaaa) [reply mobile/src/screens/meditation-home/libs/constants.ts]: @danilittaa mockedData was written like that thats why im asking ) and i saw examples for  B22 : 
const MAX_HEIGHT = 6996; 
const IDX_NOT_FOUND = -1;   
so i just thought array is not related to this 🤔 . because for arrays we have B18. Arrays are named plural nouns. (and in examples: 
const ages = [10, 15, 22];
const names = ['John', 'Pit', 'Brew'];

const cat = {
  name: 'Pit',
  friends: ['Nike', 'Sof', 'Kat'],
};

**Comment by danilittaa** [comment]
@v0ldemar01 , Could we please leave the duration mocked for now because I really need this pr to be merged. I will fix the duration in the pr [cp-182: add meditation timer](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/388#top)
#388 later

File: mobile/src/libs/components/icon/libs/maps/icon-name-to-icon/icon-name-to-icon.map.ts [hunk 4]
```diff
@@ -1,40 +1,40 @@
...
+import Forward from '#assets/img/icons/forward.svg';
+import Home from '#assets/img/icons/home.svg';
+import Next from '#assets/img/icons/next.svg';
+import Pause from '#assets/img/icons/pause.svg';
+import Play from '#assets/img/icons/play.svg';
+import Plus from '#assets/img/icons/plus.svg';
+import Previous from '#assets/img/icons/previous.svg';
+import SignOut from '#assets/img/icons/sign-out.svg';
+import Upload from '#assets/img/icons/upload.svg';
+import User from '#assets/img/icons/user.svg';
```
Comment (danilittaa) [inline mobile/src/libs/components/icon/libs/maps/icon-name-to-icon/icon-name-to-icon.map.ts]: Why did you change this? Liza said it should be in lowercase

**Review by danilittaa — CHANGES_REQUESTED** [review]

File: mobile/src/libs/components/icon/libs/maps/icon-name-to-icon/icon-name-to-icon.map.ts (hunk 4 above)
Comment (iamAlinaaa) [reply mobile/src/libs/components/icon/libs/maps/icon-name-to-icon/icon-name-to-icon.map.ts]: @danilittaa  because i have delete icon, and writing just 'delete' is forbidden in code

Comment (danilittaa) [reply mobile/src/libs/components/icon/libs/maps/icon-name-to-icon/icon-name-to-icon.map.ts]: rename it to trashBox

File: mobile/src/screens/meditation-home/meditation-home.tsx [hunk 5]
```diff
@@ -64,7 +53,9 @@ const MeditationHome: React.FC = () => {
             return (
               <Card
                 title={item.name}
-                onPress={handleSelectMeditation}
+                onPress={(): void => {
+                  handleSelectMeditation(item.name);
+                }}
```
Comment (dmytro-kucherenko) [inline mobile/src/screens/meditation-home/meditation-home.tsx]:
```suggestion
                onPress={handleSelectMeditation(item.name)}}
```

Lets use it directly, as it's our approach in the project:

```
const handleSelectMeditation = useCallback(
  (title: string) => {
    return () => {
      navigation.navigate(ChatScreenName.MEDITATION_LIST, { title });
    };
  },
  [navigation],
);
```

Comment (iamAlinaaa) [reply mobile/src/screens/meditation-home/meditation-home.tsx]: @Dmytro-Kucherenko i cant use it directly, i was said to do it like this, to avoid extra props

Comment (dmytro-kucherenko) [reply mobile/src/screens/meditation-home/meditation-home.tsx]: @iamAlinaaa Won't method with currying functions I described work fine, cause it returns another function? It's our approach in web, correct me if not

Comment (liza-veis) [reply mobile/src/screens/meditation-home/meditation-home.tsx]: we should avoid returning function from function, it complicates the logic

Comment (Liza-Veis) [reply mobile/src/screens/meditation-home/meditation-home.tsx:58]: as i see on frontend we don't return function but we pass title inside the component, which is not a good approach, we will need to fix this

File: mobile/src/screens/meditation-home/libs/constants/constants.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+const navigationItems = [
```
Comment (v0ldemar01) [inline mobile/src/screens/meditation-home/libs/constants/constants.ts]: Quality criterial rule is broken

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

**Review by v0ldemar01 — APPROVED** [review]

**Review by fshabanov — APPROVED** [review]
